File: benchmarks/poly/matrixmul/matrixmul.rb
def main
  _t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
  n = 200
  a = Array.new(n) { |i| Array.new(n) { |j| ((i * n + j) % 13).to_f } }
  b = Array.new(n) { |i| Array.new(n) { |j| ((j * n + i) % 17).to_f } }
  c = Array.new(n) { Array.new(n, 0.0) }

  n.times do |i|
    n.times do |j|
      s = 0.0
      n.times do |k|
        s += a[i][k] * b[k][j]
      end
      c[i][j] = s
    end
  end

  checksum = 0.0
  n.times do |i|
    n.times do |j|
      checksum += c[i][j]
    end
  end

  puts checksum.to_i
  $stderr.puts "bench_ns=#{Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond) - _t0}"
end

main

File: benchmarks/poly/matrixmul/matrixmul.pl
use strict;
use warnings;
use Time::HiRes qw(clock_gettime CLOCK_MONOTONIC);

sub main {
    my $_t0 = clock_gettime(CLOCK_MONOTONIC);
    my $n = 200;
    my (@a, @b, @c);
    for (my $i = 0; $i < $n; $i++) {
        for (my $j = 0; $j < $n; $j++) {
            $a[$i][$j] = ($i * $n + $j) % 13;
            $b[$i][$j] = ($j * $n + $i) % 17;
            $c[$i][$j] = 0.0;
        }
    }

    for (my $i = 0; $i < $n; $i++) {
        for (my $j = 0; $j < $n; $j++) {
            my $s = 0.0;
            for (my $k = 0; $k < $n; $k++) {
                $s += $a[$i][$k] * $b[$k][$j];
            }
            $c[$i][$j] = $s;
        }
    }

    my $checksum = 0.0;
    for (my $i = 0; $i < $n; $i++) {
        for (my $j = 0; $j < $n; $j++) {
            $checksum += $c[$i][$j];
        }
    }

    print int($checksum), "\n";
    printf STDERR "bench_ns=%.0f\n", (clock_gettime(CLOCK_MONOTONIC) - $_t0) * 1e9;
}

main();
